editor/src/core/vscode/vscode-bridge.ts

```diff
@@ -71,7 +71,9 @@ async function writeProjectFile(
       return ensureDirectoryExists(toFSPath(projectPath))
     }
     case 'TEXT_FILE': {
-      return writeFileAsUTF8(toFSPath(projectPath), file.fileContents.code)
+      const savedContent = file.lastSavedContents?.code ?? file.fileContents.code
```

Comment:
minor suggestion: this could use the `getSavedCodeFromTextFile` helper

Reply:
Good catch! Done :+1: